pkg/generate.go

```diff
@@ -407,6 +438,61 @@ func generateService(options gitopsv1alpha1.GeneratorOptions) *corev1.Service {
 	return &service
 }
 
+func generateIngress(options gitopsv1alpha1.GeneratorOptions) *networkingv1.Ingress {
+
+	ingressName := options.Name
+	if len(ingressName) >= 30 {
+		ingressName = ingressName[0:25] + util.GetRandomString(4, true)
+	}
+	k8sLabels := generateK8sLabels(options)
+
+	implementationSpecific := networkingv1.PathTypeImplementationSpecific
+
+	ingress := networkingv1.Ingress{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Ingress",
+			APIVersion: "networking.k8s.io/v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      ingressName,
+			Namespace: options.Namespace,
+			Labels:    k8sLabels,
+		},
+		Spec: networkingv1.IngressSpec{
+			Rules: []networkingv1.IngressRule{
+				{
+					IngressRuleValue: networkingv1.IngressRuleValue{
+						HTTP: &networkingv1.HTTPIngressRuleValue{
+							Paths: []networkingv1.HTTPIngressPath{
+								{
+									Path:     "/",
+									PathType: &implementationSpecific,
+									Backend: networkingv1.IngressBackend{
+										Service: &networkingv1.IngressServiceBackend{
+											Name: options.Name,
+											Port: networkingv1.ServiceBackendPort{
+												Number: int32(options.TargetPort),
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if options.Route != "" {
+		if len(ingress.Spec.Rules) > 0 {
```

Comment:
What do you think about using:
`if options.Route != "" && len(ingress.Spec.Rules) > 0 {`